share/p2p/discovery/manager.go

```diff
@@ -0,0 +1,49 @@
+package discovery
```

Comment:
I didn't know where else to leave this comment so I'll do it here: 

Please note that there is one discovery parameters that is shared between both archival + full discovery instances. Right now I don't want to manage several configs per discovery type so i'm just going to leave it as is, but it's just something to note that both instance of discovery are tied to the same configuration options (minus `EnableAdvertise` bc I configure that per instance in nodebuilder).

Reply:
Also the separation between Parameters + options in discovery is kind of ugly. I propose to clean this up in later PRs as tech debt otherwise this PR will continue to grow in scope. 

(e.g. I'd like to clean up the way onPeersUpdate works so that I don't have to do annoying magic with fx to get it set up for archival peer manager)

Reply:
I agree with Vlad on this point. I would be comfortable shipping like this given that you or someone else already commits to working on it next

Reply:
Which point tho?